application/src/main/java/uk/nhs/hee/tis/revalidation/service/DoctorsForDBService.java

```diff
@@ -156,7 +164,19 @@ public void updateDoctorConnection(final ConnectionMessageDto message) {
       doctorsForDb.setGmcLastUpdatedDateTime(message.getGmcLastUpdatedDateTime());
       doctorsRepository.save(doctorsForDb);
     } else {
-      log.info("No doctor found to update designated body code");
+      log.info(
+          "No doctor found to update for doctor GMC Number: %s, DBC: %s, creating partial record");
+      DoctorsForDB partialDoctorRecord = DoctorsForDB.builder()
+          .gmcReferenceNumber(message.getGmcId())
+          .submissionDate(message.getSubmissionDate())
+          .designatedBodyCode(message.getDesignatedBodyCode())
+          .gmcLastUpdatedDateTime(message.getGmcLastUpdatedDateTime())
+          .lastUpdatedDate(LocalDate.now())
```

Comment:
I started trying to confirm items for subsequent refactoring but it's not clear why this is being set here but not above.  Given an earlier commit (7790aa5cebc1733f5859cf4c2366ba80b58502da), this may be inappropriate.